aws/sdk/examples/telephone-game/src/main.rs

```diff
@@ -0,0 +1,439 @@
+            };
+
+            // Clear last iteration's line so we can receive new user input
+            line.clear();
+        }
+    })
+}
+
+
+/**
+# Play a game of Telephone w/ AWS
+
+1. Fetch user any user input that will override default values
+1. Create a config and required clients for AWS services
+1. Create a bucket to store audio and transcriptions if none exists
+1. Run in a loop
+    1. Start a speech synthesis task and set it to output to the previously created S3 bucket
+    1. Wait up to TASK_TIMEOUT_IN_SECONDS seconds for synthesis task to complete
+        - The status of the task is checked every TASK_WAIT_INTERVAL_IN_SECONDS in a loop
+        - Break out of the loop once the task succeeds or fails
+    1. Delete any pre-existing transcription job with the name "telephone-game-transcription"
+        - Job names must be unique so we clear the old job to reuse the name.
+    1. Wait up to TASK_TIMEOUT_IN_SECONDS seconds for transcripion job to complete
+        - The status of the job is checked every TASK_WAIT_INTERVAL_IN_SECONDS in a loop
+        - Break out of the loop once the job succeeds or fails
+    1. Download the transcript JSON file from S3 and print what Transcribe heard
+    1. Break out of the loop if an error occurs or we reach the iteration limit
+1. Log a final message before closing
+    - If everything succeeded, log the original phrase, the current phrase, and how many iterations occurred
+    - If a failure occurs, log the error and any errors that caused it
+*/
+fn play_telephone(matches: &ArgMatches) {
+    let rt = Runtime::new().expect("failed to create an async runtime");
+    let number_of_iterations = matches
+        .value_of("iterations")
+        .and_then(|i| i.parse::<u32>().ok())
+        .unwrap_or(DEFAULT_NUMBER_OF_ITERATIONS);
+
+    if number_of_iterations == 0 {
+        error!("Iterations must be a number greater than 0");
+        return;
+    }
+
+    let original_phrase = matches.value_of("phrase").unwrap_or_default();
+    let mut current_phrase = original_phrase.to_owned();
+
+    let bucket_name = matches
+        .value_of("s3_bucket_name")
+        .map(ToOwned::to_owned)
+        .unwrap_or_else(|| "telephone-game".to_owned());
+
+    let result = rt.block_on(async {
+        let config = aws_config::load_from_env().await;
+        let s3_client = aws_sdk_s3::Client::new(&config);
+        let polly_client = aws_sdk_polly::Client::new(&config);
+        let transcribe_client = aws_sdk_transcribestreaming::Client::new(&config);
+        let bucket_name = create_s3_bucket_if_not_exists(&s3_client, &bucket_name)
+            .await
+            .context("Failed to complete necessary setup")?;
+
+        let mut i = 0;
+        'game: loop {
+            debug!(
+                "starting speech synthesis task for phrase '{}'",
+                &current_phrase
+            );
+
+            let mut synthesis_task = polly_client
+                .start_speech_synthesis_task()
```

Comment:
I think you can use the synthesize_speech API instead of the task API which doesn't require polling. Should be an example in the examples folder

Reply:
`start_transcription_job` needs a `Media` object as input and when creating it, it didn't even occur to me to set it's URI to anything other than S3. It needs something hosted on the web right? I couldn't use a local file without first uploading it (or so I think.)